tc/sgx/trusted_worker_manager/enclave/signup_enclave_common.cpp

```diff
@@ -0,0 +1,148 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+
+#include "enclave_t.h"
+
+#include "avalon_sgx_error.h"
+#include "zero.h"
+#include "enclave_data.h"
+#include "enclave_utils.h"
+#include "signup_enclave_common.h"
+
+
+// Initializing singleton class object which gets initialized when
+// getInstance is called
+EnclaveData* EnclaveData::instance = 0;
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_CalculateSealedEnclaveDataSize(size_t* pSealedEnclaveDataSize) {
+    tcf_err_t result = TCF_SUCCESS;
+
+    try {
+        tcf::error::ThrowIfNull(pSealedEnclaveDataSize, "Sealed signup data size pointer is NULL");
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        *pSealedEnclaveDataSize = enclaveData->get_sealed_data_size();
+    } catch (tcf::error::Error& e) {
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Error in Avalon enclave(ecall_CalculateSealedEnclaveDataSize): %04X -- %s",
+            e.error_code(), e.what());
+        ocall_SetErrorMessage(e.what());
+        result = e.error_code();
+    } catch (...) {
+        SAFE_LOG(
+            TCF_LOG_ERROR, "Unknown error in Avalon enclave(ecall_CalculateSealedEnclaveDataSize)");
+        result = TCF_ERR_UNKNOWN;
+    }
+
+    return result;
+}  // ecall_CalculateSealedEnclaveDataSize
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_CalculatePublicEnclaveDataSize(size_t* pPublicEnclaveDataSize) {
+    tcf_err_t result = TCF_SUCCESS;
```

Comment:
Preferable to start initialize with failure.

Reply:
Agree, this needs to be followed in all CPP functions. I will make the changes here and rest of the places can be handled as and when we touch the code.